src/lib/mina_base/transaction_status.ml

```diff
@@ -156,8 +211,31 @@ module Failure = struct
         "The source of a signed command cannot be a snapp account"
     | Snapp_account_not_present ->
         "A snapp account does not exist"
-    | Update_not_permitted ->
-        "An account is not permitted to make the given update"
+    | Update_not_permitted_balance ->
+        "An account is not permitted to update its balance"
```

Comment:
The wording might be clearer as `The authentication for an account didn't allow the requested update to its balance` or similar, and likewise below.